File: BAU_Automation_Tests_2/BauAutomationTests/src/test/java/com/wh/auto/nova21/desktop/pages/DepositPage.java
package com.wh.auto.nova21.desktop.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

import com.wh.auto.TestObjects.AbstractWhPageObject;
import com.wh.auto.common.Browsers;
import com.wh.auto.common.WebDriverUtils;

public class DepositPage extends AbstractWhPageObject{

	@FindBy(css = "#RC_Submit > tbody > tr > td.main > span")
	private WebElement submit;
	//private By submit = By.cssSelector("#RC_Submit > tbody > tr > td.main > span");
	@FindBy(css= "#DCC_Submit > tbody > tr > td.main > span")
	private WebElement depositButten;
	//private By depositButten = By.cssSelector("#DCC_Submit > tbody > tr > td.main > span");
	private By deposit = By.xpath("//div[@id='column-6']/div[3]/a/span");
	//#RC_Submit > tbody > tr > td.main > span
	private String month;
	
	public DepositPage(Browsers browsers) { 
		   super(browsers);
	    }
	
	public void deposit() throws InterruptedException {
		//driver = browsers.get_Driver();
		waitForElement(driver, deposit, 15).click();//deposit
		Thread.sleep(5000);
		waitForElement(driver, By.id("cashier-frame"), 30);
		Thread.sleep(5000);
		driver.switchTo().frame(driver.findElement(By.id("cashier-frame")));
		driver.findElement(By.id("RC_Amount")).clear();
		driver.findElement(By.id("RC_Amount")).sendKeys("");
		new Select(driver.findElement(By.id("RC_CardType"))).selectByVisibleText("Visa Delta");
		driver.findElement(By.id("RC_CardNumber")).clear();
		driver.findElement(By.id("RC_CardNumber")).sendKeys("1111111111111111");
		new Select(driver.findElement(By.id("RC_ExpirationMonth"))).selectByVisibleText(getMonth());//01
		new Select(driver.findElement(By.id("RC_ExpirationYear"))).selectByVisibleText("2024");
		driver.findElement(By.id("RC_CVV2")).clear();
		driver.findElement(By.id("RC_CVV2")).sendKeys("111");
		Thread.sleep(1000);
		submit.click();//submit
		new WebDriverWait(driver, 35).until(ExpectedConditions.visibilityOf(
		driver.findElement(By.id("UM_Buttons")))).click();//OK button
		//new WebDriverWait(driver, 10).until(ExpectedConditions.visibilityOf(
		//driver.findElement(By.id("UM_OKButtonText")))).click();//OK button
	}
	
	public String depositNext() throws InterruptedException {//100 111
		//driver = browsers.get_Driver();
		//driver.switchTo().frame(driver.findElement(By.id("cashier-frame")));
		waitForElement(driver, By.id("D_Amount"), 25).sendKeys("100");
		waitForElement(driver, By.id("DCC_CVV2"), 25).sendKeys("111");
		Thread.sleep(1000);
		//new WebDriverWait(driver, 25).until(ExpectedConditions.visibilityOf(
		//		driver.findElement(By.cssSelector("#DCC_Submit > tbody > tr > td.main > span")))).click();//submit
		depositButten.click();
		waitForElement(driver, By.id("depositConfirmed"), 25).click();//confirm Deposit radio
		waitForElement(driver, By.id("UM_OKButtonText"), 10).click();//confirm butten
		driver.switchTo().defaultContent();
		try {waitForElement(driver, By.cssSelector("button.ok"), 45).click();
			waitForElement(driver, By.cssSelector("button.ok"), 45).click();
			Thread.sleep(1000);
			waitForElement(driver, By.cssSelector("button.ok"), 25).click();
		} catch (Exception e) {
			//e.printStackTrace();
		}
		//String bonusMsg = WebDriverUtils.waitForElement(driver, By.cssSelector(".message-area"), 25).getText();
		waitForElement(driver, By.cssSelector(".popupClose"), 25).click();
		return null;
	}

	private String getMonth() {
		return month;
	}

	public void setMonth(String month) {
		this.month = month;
	}
	
}

File: BAU_Automation_Tests_2/ClientsAutomationTests/EurograndWebEn/DepositPage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using AutomationLib;
using System.Windows.Automation;
using System.Threading;
using System.Windows.Forms;
using System.Windows;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;

namespace EurograndWebEn
{
	/// <summary>
	/// The Deposit Page
	/// </summary>
	public class DepositPage : Automation_Lib
	{
		private readonly IWebDriver _driver;

		public DepositPage(IWebDriver driver)
		{
			_driver = driver;
		}

		/// <summary>
		/// Deposit 50 URO
		/// </summary>
		/// <returns></returns>
		public void Deposit()
		{
			//driver.SwitchTo().Frame("innerIframe");
			WebSelenium.FindElement(_driver, By.Id("D_Amount"), 20);
			_driver.FindElement(By.Id("D_Amount")).Clear();
			_driver.FindElement(By.Id("D_Amount")).SendKeys("100");
			_driver.FindElement(By.Id("DCC_CVV2")).Clear();
			_driver.FindElement(By.Id("DCC_CVV2")).SendKeys("111");
			//FindElement(By.CssSelector("td.main > span"), 5).Click();
			//click on deposit
			Condition controlCondition = new PropertyCondition(AutomationElement.NameProperty, "Registration Done");
			AutomationElement mainWindow = waitforWindowNameClass("Registration Done - Google Chrome", controlCondition, 30, true);
			Rect rect = mainWindow.Current.BoundingRectangle;
			System.Drawing.Point p = Cursor.Position;
			int x = Convert.ToInt32(rect.Left) + 690;//690, 358
			int y = Convert.ToInt32(rect.Top) + 382;
			MouseClick(x, y);
		}

		/// <summary>
		/// confirmDeposit
		/// </summary>
		public void confirmDeposit()
		{
			//new WebDriverWait(_driver, 25).until(ExpectedConditions.visibilityOf(
			//	_driver.findElement(By.cssSelector("#DCC_Submit > tbody > tr > td.main > span"))))
			WebSelenium.FindElement(_driver, By.Id("depositConfirmed"), 25).Click();//confirm Deposit radio
			WebSelenium.FindElement(_driver, By.Id("UM_OKButtonText"), 10).Click();//confirm butten
			//waitForElement(driver, By.id("depositConfirmed"), 25).click();//confirm Deposit radio
			//waitForElement(driver, By.id("UM_OKButtonText"), 10).click();//confirm butten
			//Condition controlNameCondition = new PropertyCondition(AutomationElement.LocalizedControlTypeProperty, "check box");
			//AutomationElement mainWindow2 = waitforWindowNameClass("PTIODEVICE", controlNameCondition, 60, false);
			//AutomationElement control = mainWindow2.FindFirst(TreeScope.Element | TreeScope.Descendants, controlNameCondition);
			//ClickElement(control);//PROCEED chackBox
			//Thread.Sleep(1000);
			//SendKeys.SendWait("{TAB}");
			//SendKeys.SendWait("{ENTER}");
			//controlNameCondition = new PropertyCondition(AutomationElement.NameProperty, "PROCEED");
			//control = mainWindow2.FindFirst(TreeScope.Element | TreeScope.Descendants, controlNameCondition);
			//ClickElement(control);//PROCEED button
		}

		/// <summary>
		/// Popup Your deposit has been approved and will be charged as WH ONLINE. Thank you and good luck!
		/// </summary>
		public void CloseApprovePopup()
		{
			new WebDriverWait(_driver, TimeSpan.FromSeconds(300)).
				Until(drv => (drv.FindElement(By.Id("UM_OKButtonText")).
					Displayed) ? drv.FindElement(By.Id("UM_OKButtonText")) : null).Click();
			//for (int i = 0; i < 200; i++)
			//{
			//    if (driver.FindElement(By.Id("UM_OKButtonText")).Displayed)
			//    {
			//        driver.FindElement(By.Id("UM_OKButtonText")).Click();
			//        break;
			//    }
			//    Thread.Sleep(1000);
			//}
		}

		/// <summary>
		/// Read Current Balance
		/// </summary>
		public string ReadCurrentBalance()
		{
			return WebSelenium.FindElement(_driver, By.Id("PH_UserBalance"), 5).Text;
		}

	}
}
